dev/io.openliberty.io.netty/src/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -1137,8 +1157,15 @@ private static class HeaderParser {
         public ByteBuf parse(ByteBuf buffer) {
             final int readableBytes = buffer.readableBytes();
             final int readerIndex = buffer.readerIndex();
-            final int maxBodySize = maxLength - size;
-            assert maxBodySize >= 0;
+            long maxAllowedBody;
+            // If -1 this check should be disabled
+            if (maxLength == -1) {
+            	maxAllowedBody = readableBytes - 2L;
```

Comment:
Adjust according to our peer to peer revision. Account the -2L instead by not using the maxBodySizzeWithCRLF